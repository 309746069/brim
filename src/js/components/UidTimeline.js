/* @flow */

import React from "react"
import Log from "../models/Log"
import * as TimeField from "../lib/TimeField"
import * as d3 from "d3"
import * as Time from "../lib/Time"
import * as TimeWindow from "../lib/TimeWindow"
import isEqual from "lodash/isEqual"

type Props = {
  viewLogDetail: Function,
  logs: Log[],
  currentLog: Log
}

export default class UidWaterfall extends React.Component<Props> {
  row(log: Log, index: number, xScale) {
    const ts = TimeField.toDate(log.get("ts"))
    const position = xScale(ts)
    const isCurrent = isEqual(log, this.props.currentLog)
    return (
      <div
        key={index}
        className="waterfall-row"
        onClick={() => this.props.viewLogDetail(log)}
      >
        <div className="ts-label">{Time.format(ts, "HH:mm:ss.SSS")}</div>
        <div className="slider">
          <div className="line" />
          <span
            className={`path-tag ${log.get("_path")}-bg-color ${
              isCurrent ? "current" : ""
            }`}
            style={{left: position + "%"}}
          >
            {log.get("_path")}
          </span>
        </div>
      </div>
    )
  }

  render() {
    if (this.props.logs.length === 0) return null

    const extent = d3.extent(this.props.logs.map(l => l.cast("ts")))
    const xScale = d3
      .scaleTime()
      .domain(extent)
      .range([0, 100])

    const duration = TimeWindow.duration(extent, "seconds")

    return (
      <div className="uid-waterfall">
        {this.props.logs.map((log, i) => this.row(log, i, xScale))}
        <p className="duration">Duration: {duration}s</p>
      </div>
    )
  }
}
