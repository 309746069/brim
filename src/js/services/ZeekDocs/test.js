/* @flow */
import ZeekDocs from "./"

test("describe conn uid", () => {
  let desc = ZeekDocs.describe("conn", "uid")

  expect(desc).toBe("A unique identifier of the connection.")
})
