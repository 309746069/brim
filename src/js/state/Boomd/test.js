/* @flow */
import Boomd from "./"
import initTestStore from "../../test/initTestStore"

test("setting use analytics cache", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useCache(true))
  expect(Boomd.usingCache(store.getState())).toEqual(true)
  store.dispatch(Boomd.useCache(false))
  expect(Boomd.usingCache(store.getState())).toEqual(false)
})

test("setting use index", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useIndex(true))
  expect(Boomd.usingIndex(store.getState())).toEqual(true)
  store.dispatch(Boomd.useIndex(false))
  expect(Boomd.usingIndex(store.getState())).toEqual(false)
})
