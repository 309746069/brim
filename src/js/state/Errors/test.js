/* @flow */

import AppError from "../../models/AppError"
import Errors from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("ERROR_CREATE", () => {
  let state = store.dispatchAll([Errors.createError("Bug")])

  expect(Errors.getErrors(state)[0]).toEqual(expect.any(AppError))
})

test("ERRORS_CLEAR", () => {
  let state = store.dispatchAll([
    Errors.createError("Bug"),
    Errors.createError("Bug2"),
    Errors.clearErrors()
  ])

  expect(Errors.getErrors(state)).toEqual([])
})
