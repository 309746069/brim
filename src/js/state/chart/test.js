/* @flow */

import chart from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let records = [
  [{name: "id", type: "string", value: "1"}],
  [{name: "id", type: "string", value: "2"}]
]

test("chart records append", () => {
  let state = store.dispatchAll([
    chart.appendRecords([records[0]]),
    chart.appendRecords([records[1]])
  ])

  expect(chart.getRecords(state)).toEqual(records)
})

test("chart records status", () => {
  let state = store.dispatchAll([chart.setStatus("SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  let state = store.dispatchAll([
    chart.appendRecords(records),
    chart.setStatus("SUCCESS"),
    chart.clear()
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getRecords(state)).toEqual([])
})
