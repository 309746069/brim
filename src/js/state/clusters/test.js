/* @flow */

import {addCluster, removeCluster, setCluster, setClusterError} from "./actions"
import {
  getCurrentCluster,
  getClusterMessage,
  getSavedClusters
} from "./selectors"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let cluster = {
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  let state = store.dispatchAll([addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("addCluster when it already exists", () => {
  let state = store.dispatchAll([addCluster(cluster), addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("removeCluster", () => {
  let state = store.dispatchAll([addCluster(cluster), removeCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([])
})

test("setCluster", () => {
  let state = store.dispatchAll([setCluster(cluster)])

  expect(getCurrentCluster(state)).toEqual(cluster)
})

test("setClusterMessage to message", () => {
  let state = store.dispatchAll([setClusterError("no spaces")])

  expect(getClusterMessage(state)).toEqual("no spaces")
})

test("setClusterMessage to empty", () => {
  let state = store.dispatchAll([setClusterError("")])

  expect(getClusterMessage(state)).toEqual("")
})
