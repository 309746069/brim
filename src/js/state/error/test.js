/* @flow */
import AppError from "../../models/AppError"
import error from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("init state", () => {
  let state = store.getState()
  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(false)
})

test("set", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e)])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(true)
})

test("clear", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.clear()])

  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(true)
})

test("dismiss", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.dismiss()])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(false)
})
