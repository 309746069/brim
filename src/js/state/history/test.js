/* @flow */

import {changeSearchBarInput} from "../actions"
import History from "./"
import Tab from "../tab"
import initTestStore from "../../test/initTestStore"
import submitSearch from "../../flows/submitSearch"

let store
beforeEach(() => {
  store = initTestStore()
  store.dispatchAll([
    changeSearchBarInput("first"),
    submitSearch(),
    changeSearchBarInput("second"),
    submitSearch(),
    changeSearchBarInput("third"),
    submitSearch()
  ])
})

test("pushing history", () => {
  const state = store.getState()
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("third")
})

test("moving back changes the position", () => {
  let state = store.dispatchAll([History.back()])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("second")
})

test("going forward in history", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    History.forward()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("second")
})

test("going back in history then pushing new history", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    changeSearchBarInput("fourth"),
    submitSearch()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("fourth")
})

test("back, back, push, back", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    changeSearchBarInput("fourth"),
    submitSearch(),
    History.back()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("first")
})

test("clearing history", () => {
  let state = store.dispatchAll([History.clear()])
  expect(Tab.currentEntry(state)).toBe(undefined)
})
