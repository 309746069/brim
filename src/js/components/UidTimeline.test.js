/* @flow */

import React from "react"
import {shallow} from "enzyme"
import UidTimeline from "./UidTimeline"
import * as mockLogs from "../test/mockLogs"

test("shapshot", () => {
  const wrapper = shallow(
    <UidTimeline
      currentLog={mockLogs.conn()}
      logs={[mockLogs.conn(), mockLogs.dns(), mockLogs.http()]}
      viewLogDetail={jest.fn()}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
