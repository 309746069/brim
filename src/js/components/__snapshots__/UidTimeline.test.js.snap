// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UidWaterfall
    log={
      Log {
        "descriptor": Array [
          Object {
            "name": "_td",
            "type": "int",
          },
          Object {
            "name": "_path",
            "type": "string",
          },
          Object {
            "name": "ts",
            "type": "time",
          },
          Object {
            "name": "uid",
            "type": "string",
          },
          Object {
            "name": "id.orig_h",
            "type": "addr",
          },
          Object {
            "name": "id.orig_p",
            "type": "port",
          },
          Object {
            "name": "id.resp_h",
            "type": "addr",
          },
          Object {
            "name": "id.resp_p",
            "type": "port",
          },
          Object {
            "name": "proto",
            "type": "enum",
          },
          Object {
            "name": "service",
            "type": "string",
          },
          Object {
            "name": "duration",
            "type": "interval",
          },
          Object {
            "name": "orig_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_bytes",
            "type": "count",
          },
          Object {
            "name": "conn_state",
            "type": "string",
          },
          Object {
            "name": "local_orig",
            "type": "bool",
          },
          Object {
            "name": "local_resp",
            "type": "bool",
          },
          Object {
            "name": "missed_bytes",
            "type": "count",
          },
          Object {
            "name": "history",
            "type": "string",
          },
          Object {
            "name": "orig_pkts",
            "type": "count",
          },
          Object {
            "name": "orig_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_pkts",
            "type": "count",
          },
          Object {
            "name": "resp_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "tunnel_parents",
            "type": "set[string]",
          },
        ],
        "tuple": Array [
          "0",
          "conn",
          "1425612054.369843",
          "Cynwae4qh1GxM82hQ2",
          "192.168.0.50",
          "1900",
          "239.255.255.250",
          "1900",
          "udp",
          "-",
          "2.000293",
          "282",
          "0",
          "S0",
          "-",
          "-",
          "0",
          "D",
          "3",
          "366",
          "0",
          "0",
          "(empty)",
        ],
      }
    }
    logs={
      Array [
        Log {
          "descriptor": Array [
            Object {
              "name": "_td",
              "type": "int",
            },
            Object {
              "name": "_path",
              "type": "string",
            },
            Object {
              "name": "ts",
              "type": "time",
            },
            Object {
              "name": "uid",
              "type": "string",
            },
            Object {
              "name": "id.orig_h",
              "type": "addr",
            },
            Object {
              "name": "id.orig_p",
              "type": "port",
            },
            Object {
              "name": "id.resp_h",
              "type": "addr",
            },
            Object {
              "name": "id.resp_p",
              "type": "port",
            },
            Object {
              "name": "proto",
              "type": "enum",
            },
            Object {
              "name": "service",
              "type": "string",
            },
            Object {
              "name": "duration",
              "type": "interval",
            },
            Object {
              "name": "orig_bytes",
              "type": "count",
            },
            Object {
              "name": "resp_bytes",
              "type": "count",
            },
            Object {
              "name": "conn_state",
              "type": "string",
            },
            Object {
              "name": "local_orig",
              "type": "bool",
            },
            Object {
              "name": "local_resp",
              "type": "bool",
            },
            Object {
              "name": "missed_bytes",
              "type": "count",
            },
            Object {
              "name": "history",
              "type": "string",
            },
            Object {
              "name": "orig_pkts",
              "type": "count",
            },
            Object {
              "name": "orig_ip_bytes",
              "type": "count",
            },
            Object {
              "name": "resp_pkts",
              "type": "count",
            },
            Object {
              "name": "resp_ip_bytes",
              "type": "count",
            },
            Object {
              "name": "tunnel_parents",
              "type": "set[string]",
            },
          ],
          "tuple": Array [
            "0",
            "conn",
            "1425612054.369843",
            "Cynwae4qh1GxM82hQ2",
            "192.168.0.50",
            "1900",
            "239.255.255.250",
            "1900",
            "udp",
            "-",
            "2.000293",
            "282",
            "0",
            "S0",
            "-",
            "-",
            "0",
            "D",
            "3",
            "366",
            "0",
            "0",
            "(empty)",
          ],
        },
        Log {
          "descriptor": Array [
            Object {
              "name": "_td",
              "type": "int",
            },
            Object {
              "name": "_path",
              "type": "string",
            },
            Object {
              "name": "ts",
              "type": "time",
            },
            Object {
              "name": "uid",
              "type": "string",
            },
            Object {
              "name": "id.orig_h",
              "type": "addr",
            },
            Object {
              "name": "id.orig_p",
              "type": "port",
            },
            Object {
              "name": "id.resp_h",
              "type": "addr",
            },
            Object {
              "name": "id.resp_p",
              "type": "port",
            },
            Object {
              "name": "proto",
              "type": "enum",
            },
            Object {
              "name": "trans_id",
              "type": "count",
            },
            Object {
              "name": "rtt",
              "type": "interval",
            },
            Object {
              "name": "query",
              "type": "string",
            },
            Object {
              "name": "qclass",
              "type": "count",
            },
            Object {
              "name": "qclass_name",
              "type": "string",
            },
            Object {
              "name": "qtype",
              "type": "count",
            },
            Object {
              "name": "qtype_name",
              "type": "string",
            },
            Object {
              "name": "rcode",
              "type": "count",
            },
            Object {
              "name": "rcode_name",
              "type": "string",
            },
            Object {
              "name": "AA",
              "type": "bool",
            },
            Object {
              "name": "TC",
              "type": "bool",
            },
            Object {
              "name": "RD",
              "type": "bool",
            },
            Object {
              "name": "RA",
              "type": "bool",
            },
            Object {
              "name": "Z",
              "type": "count",
            },
            Object {
              "name": "answers",
              "type": "vector[string]",
            },
            Object {
              "name": "TTLs",
              "type": "vector[interval]",
            },
            Object {
              "name": "rejected",
              "type": "bool",
            },
          ],
          "tuple": Array [
            "8",
            "dns",
            "1425565514.419939",
            "Cum4LVba3W3KNG6qa",
            "fe80::eef4:bbff:fe51:89ec",
            "5353",
            "ff02::fb",
            "5353",
            "udp",
            "0",
            "0.119484",
            "_ipp._tcp.local",
            "1",
            "C_INTERNET",
            "12",
            "PTR",
            "0",
            "NOERROR",
            "T",
            "F",
            "F",
            "F",
            "0",
            "_workstation._tcp.local,sniffer [ec:f4:bb:51:89:ec]._workstation._tcp.local",
            "4500.000000,4500.000000",
            "F",
          ],
        },
        Log {
          "descriptor": Array [
            Object {
              "name": "_td",
              "type": "int",
            },
            Object {
              "name": "_path",
              "type": "string",
            },
            Object {
              "name": "ts",
              "type": "time",
            },
            Object {
              "name": "uid",
              "type": "string",
            },
            Object {
              "name": "id.orig_h",
              "type": "addr",
            },
            Object {
              "name": "id.orig_p",
              "type": "port",
            },
            Object {
              "name": "id.resp_h",
              "type": "addr",
            },
            Object {
              "name": "id.resp_p",
              "type": "port",
            },
            Object {
              "name": "trans_depth",
              "type": "count",
            },
            Object {
              "name": "method",
              "type": "string",
            },
            Object {
              "name": "host",
              "type": "string",
            },
            Object {
              "name": "uri",
              "type": "string",
            },
            Object {
              "name": "referrer",
              "type": "string",
            },
            Object {
              "name": "version",
              "type": "string",
            },
            Object {
              "name": "user_agent",
              "type": "string",
            },
            Object {
              "name": "request_body_len",
              "type": "count",
            },
            Object {
              "name": "response_body_len",
              "type": "count",
            },
            Object {
              "name": "status_code",
              "type": "count",
            },
            Object {
              "name": "status_msg",
              "type": "string",
            },
            Object {
              "name": "info_code",
              "type": "count",
            },
            Object {
              "name": "info_msg",
              "type": "string",
            },
            Object {
              "name": "tags",
              "type": "set[enum]",
            },
            Object {
              "name": "username",
              "type": "string",
            },
            Object {
              "name": "password",
              "type": "string",
            },
            Object {
              "name": "proxied",
              "type": "set[string]",
            },
            Object {
              "name": "orig_fuids",
              "type": "vector[string]",
            },
            Object {
              "name": "orig_filenames",
              "type": "vector[string]",
            },
            Object {
              "name": "orig_mime_types",
              "type": "vector[string]",
            },
            Object {
              "name": "resp_fuids",
              "type": "vector[string]",
            },
            Object {
              "name": "resp_filenames",
              "type": "vector[string]",
            },
            Object {
              "name": "resp_mime_types",
              "type": "vector[string]",
            },
          ],
          "tuple": Array [
            "10",
            "http",
            "1425567042.047800",
            "CBQjjR1ZrY6LmUfzX5",
            "192.168.0.51",
            "60677",
            "91.189.89.240",
            "80",
            "1",
            "GET",
            "start.ubuntu.com",
            "/14.10/Google/?sourceid=hp",
            "-",
            "1.0",
            "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:33.0) Gecko/20100101 Firefox/33.0",
            "0",
            "6005",
            "200",
            "OK",
            "-",
            "-",
            "(empty)",
            "-",
            "-",
            "-",
            "-",
            "-",
            "-",
            "FUPWLQXTNsTNvf33",
            "-",
            "text/html",
          ],
        },
      ]
    }
    viewLogDetail={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="waterfall-row"
            onClick={[Function]}
          >
            <div
              className="ts-label"
            >
              19:20:54.369
            </div>
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag conn-bg-color "
                style={
                  Object {
                    "left": "100%",
                  }
                }
              >
                conn
              </span>
            </div>
          </div>,
          <div
            className="waterfall-row"
            onClick={[Function]}
          >
            <div
              className="ts-label"
            >
              06:25:14.419
            </div>
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag dns-bg-color "
                style={
                  Object {
                    "left": "0%",
                  }
                }
              >
                dns
              </span>
            </div>
          </div>,
          <div
            className="waterfall-row"
            onClick={[Function]}
          >
            <div
              className="ts-label"
            >
              06:50:42.047
            </div>
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag http-bg-color "
                style={
                  Object {
                    "left": "3.2824014636887235%",
                  }
                }
              >
                http
              </span>
            </div>
          </div>,
        ],
        <p
          className="duration"
        >
          Duration: 
          46539.95
          s
        </p>,
      ],
      "className": "uid-waterfall",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ts-label"
            >
              19:20:54.369
            </div>,
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag conn-bg-color "
                style={
                  Object {
                    "left": "100%",
                  }
                }
              >
                conn
              </span>
            </div>,
          ],
          "className": "waterfall-row",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "19:20:54.369",
              "className": "ts-label",
            },
            "ref": null,
            "rendered": "19:20:54.369",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="line"
                />,
                <span
                  className="path-tag conn-bg-color "
                  style={
                    Object {
                      "left": "100%",
                    }
                  }
                >
                  conn
                </span>,
              ],
              "className": "slider",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "conn",
                  "className": "path-tag conn-bg-color ",
                  "style": Object {
                    "left": "100%",
                  },
                },
                "ref": null,
                "rendered": "conn",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ts-label"
            >
              06:25:14.419
            </div>,
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag dns-bg-color "
                style={
                  Object {
                    "left": "0%",
                  }
                }
              >
                dns
              </span>
            </div>,
          ],
          "className": "waterfall-row",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "06:25:14.419",
              "className": "ts-label",
            },
            "ref": null,
            "rendered": "06:25:14.419",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="line"
                />,
                <span
                  className="path-tag dns-bg-color "
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                >
                  dns
                </span>,
              ],
              "className": "slider",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dns",
                  "className": "path-tag dns-bg-color ",
                  "style": Object {
                    "left": "0%",
                  },
                },
                "ref": null,
                "rendered": "dns",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ts-label"
            >
              06:50:42.047
            </div>,
            <div
              className="slider"
            >
              <div
                className="line"
              />
              <span
                className="path-tag http-bg-color "
                style={
                  Object {
                    "left": "3.2824014636887235%",
                  }
                }
              >
                http
              </span>
            </div>,
          ],
          "className": "waterfall-row",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "06:50:42.047",
              "className": "ts-label",
            },
            "ref": null,
            "rendered": "06:50:42.047",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="line"
                />,
                <span
                  className="path-tag http-bg-color "
                  style={
                    Object {
                      "left": "3.2824014636887235%",
                    }
                  }
                >
                  http
                </span>,
              ],
              "className": "slider",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "http",
                  "className": "path-tag http-bg-color ",
                  "style": Object {
                    "left": "3.2824014636887235%",
                  },
                },
                "ref": null,
                "rendered": "http",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Duration: ",
            46539.95,
            "s",
          ],
          "className": "duration",
        },
        "ref": null,
        "rendered": Array [
          "Duration: ",
          46539.95,
          "s",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="waterfall-row"
              onClick={[Function]}
            >
              <div
                className="ts-label"
              >
                19:20:54.369
              </div>
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag conn-bg-color "
                  style={
                    Object {
                      "left": "100%",
                    }
                  }
                >
                  conn
                </span>
              </div>
            </div>,
            <div
              className="waterfall-row"
              onClick={[Function]}
            >
              <div
                className="ts-label"
              >
                06:25:14.419
              </div>
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag dns-bg-color "
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                >
                  dns
                </span>
              </div>
            </div>,
            <div
              className="waterfall-row"
              onClick={[Function]}
            >
              <div
                className="ts-label"
              >
                06:50:42.047
              </div>
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag http-bg-color "
                  style={
                    Object {
                      "left": "3.2824014636887235%",
                    }
                  }
                >
                  http
                </span>
              </div>
            </div>,
          ],
          <p
            className="duration"
          >
            Duration: 
            46539.95
            s
          </p>,
        ],
        "className": "uid-waterfall",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ts-label"
              >
                19:20:54.369
              </div>,
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag conn-bg-color "
                  style={
                    Object {
                      "left": "100%",
                    }
                  }
                >
                  conn
                </span>
              </div>,
            ],
            "className": "waterfall-row",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "19:20:54.369",
                "className": "ts-label",
              },
              "ref": null,
              "rendered": "19:20:54.369",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="line"
                  />,
                  <span
                    className="path-tag conn-bg-color "
                    style={
                      Object {
                        "left": "100%",
                      }
                    }
                  >
                    conn
                  </span>,
                ],
                "className": "slider",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "conn",
                    "className": "path-tag conn-bg-color ",
                    "style": Object {
                      "left": "100%",
                    },
                  },
                  "ref": null,
                  "rendered": "conn",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ts-label"
              >
                06:25:14.419
              </div>,
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag dns-bg-color "
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                >
                  dns
                </span>
              </div>,
            ],
            "className": "waterfall-row",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "06:25:14.419",
                "className": "ts-label",
              },
              "ref": null,
              "rendered": "06:25:14.419",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="line"
                  />,
                  <span
                    className="path-tag dns-bg-color "
                    style={
                      Object {
                        "left": "0%",
                      }
                    }
                  >
                    dns
                  </span>,
                ],
                "className": "slider",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "dns",
                    "className": "path-tag dns-bg-color ",
                    "style": Object {
                      "left": "0%",
                    },
                  },
                  "ref": null,
                  "rendered": "dns",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ts-label"
              >
                06:50:42.047
              </div>,
              <div
                className="slider"
              >
                <div
                  className="line"
                />
                <span
                  className="path-tag http-bg-color "
                  style={
                    Object {
                      "left": "3.2824014636887235%",
                    }
                  }
                >
                  http
                </span>
              </div>,
            ],
            "className": "waterfall-row",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "06:50:42.047",
                "className": "ts-label",
              },
              "ref": null,
              "rendered": "06:50:42.047",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="line"
                  />,
                  <span
                    className="path-tag http-bg-color "
                    style={
                      Object {
                        "left": "3.2824014636887235%",
                      }
                    }
                  >
                    http
                  </span>,
                ],
                "className": "slider",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "http",
                    "className": "path-tag http-bg-color ",
                    "style": Object {
                      "left": "3.2824014636887235%",
                    },
                  },
                  "ref": null,
                  "rendered": "http",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Duration: ",
              46539.95,
              "s",
            ],
            "className": "duration",
          },
          "ref": null,
          "rendered": Array [
            "Duration: ",
            46539.95,
            "s",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
